File: 滑动窗口/209minSubArrayLen.js
function minSubArrayLen(target, nums) {
  let sum = 0;
  let l = 0;
  let r = 0;
  let len = nums.length;
  let res = len + 1;
  while (r < len) {
    sum += nums[r];
    r += 1;
    while (sum >= target) {
      res = res < r - l ? res : r - l;
      sum -= nums[l];
      l += 1;
    }
  }
  return res > len ? 0 : res;
}

File: 滑动窗口/209minSubArrayLen.go
package main

func min(a int, b int) int {
	if a < b {
		return a
	} else {
		return b
	}
}

func minSubArrayLen(target int, nums []int) int {
	length := len(nums)
	res := length + 1
	i, j, sum := 0, 0, 0
	// 滑动窗口
	for j < length {
		sum += nums[j]
		j++
		for sum >= target {
			res = min(res, j-i)
			sum -= nums[i]
			i++
		}
	}
	if res == length+1 {
		return 0
	} else {
		return res
	}
}

func main() {
	res := minSubArrayLen(11, []int{1, 2, 3, 4, 5})
	print(res)
}
